jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorization.java

```diff
@@ -34,7 +44,57 @@
    * @return an {@link Authorization} with a {@code Bearer} token
    */
   public static Authorization fromBearerToken(String token) {
-    return new Authorization("Bearer", token);
+    return new Authorization("Bearer", token, decodeTokenRepositoryGrants(token));
+  }
+
+  /**
+   * Decode the <a href="https://docs.docker.com/registry/spec/auth/jwt/">Docker Registry v2 Bearer
+   * Token</a> to list the granted repositories with their levels of access.
+   *
+   * @param token a Docker Registry Bearer Token
+   * @return a mapping of repository to granted access scopes, or {@code null} if there is no JWT
+   *     the token is not a Docker Registry Bearer Token
+   */
+  @VisibleForTesting
+  @Nullable
+  static Multimap<String, String> decodeTokenRepositoryGrants(String token) {
+    // Docker Registry Bearer Tokens are based on JWT.  A valid JWT is a set of 3 base64-encoded
+    // parts (header, payload, signature), collated with a ".".  The header and payload are
+    // JSON objects.
+    String[] jwtParts = token.split("\\.", -1);
+    byte[] payloadData;
+    if (jwtParts.length != 3 || (payloadData = Base64.decodeBase64(jwtParts[1])) == null) {
+      return null;
+    }
+
+    // The payload looks like:
+    // {
+    //   "access":[{"type":"repository","name":"repository/name","actions":["pull"]}],
+    //   "aud":"registry.docker.io",
+    //   "iss":"auth.docker.io",
+    //   "exp":999,
+    //   "iat":999,
+    //   "jti":"zzzz",
+    //   "nbf":999,
+    //   "sub":"e3ae001d-xxx"
+    // }
+    //
+    TokenPayloadTemplate payload;
```

Comment:
This method doesn't return `IOException`, so I think the code looks aesthetically better if

```java
try {
  TokenPayloadTemplate payload = ...
  if (payload.access == null) {
    return null;
  }
  return payload.access...
} catch (IOException ex) {
  return null;
}
```